### ISSUE TITLE
Flash of white when calling setFullScreen for a window immediately after it is created or before it is shown

### ISSUE THREAD

**Issue by iamdriz** [atom/electron#1055]
When an app first loads regardless of it being set to transparent or frameless, the window is white for a second before showing the application. Anyway to stop this?

If you set the app to fullscreen on OS X, you see the white background BEFORE you see the OS X gray background you see for fullscreen windows.

**Comment by zcbenz** [comment]
People usually make the window hidden at first and then show the window when the page is fully loaded.

**Comment by iamdriz** [comment]
How do you do that? As when I set the show to false and then set it to show on app ready I see the White when the window is shown... So it's like the window is white when first shown rather than any load issue.

Also the hide and show options and methods don't work for fullscreen windows so you can't use them for this bug.

**Comment by zcbenz** [comment]
You should show the window when page is ready (namely `document.onready`), not when app is ready.

> Also the hide and show options and methods don't work for fullscreen windows so you can't use them for this bug.

You can call the `setFullScreen` API after the page is ready.

**Comment by iamdriz** [comment]
I've tried this, but I get the white screen?

In main.js
~~~
mainWindow = new BrowserWindow({
	  'width': size.width,
	  'height': size.height,
	  'frame': false,
	  'transparent': true,
	  'show': false,
	  'resizable': false,
	  'auto-hide-menu-bar': true
	});
~~~

and then in my index.html

~~~
document.addEventListener("DOMContentLoaded", function(event) { 
	showWindow();
});

function showWindow()
{
	remote.getCurrentWindow().setFullScreen(true);
}
~~~

While I'm waiting for the showWindow function to fire the whole screen is white!

**Comment by zcbenz** [comment]
It seems that calling `setFullScreen` for a window immediately after it is created or before it is shown will make the window white for a while before everything shows.

**Comment by zcbenz** [comment]
@iamdriz Can you try if showing the window before setting it to fullscreen works?

```js
function showWindow()
{
    remote.getCurrentWindow().show();
    remote.getCurrentWindow().setFullScreen(true);
}
```

**Comment by iamdriz** [comment]
Yeah I tried that. It makes it sense haha. It flashes and then you see the animation to fullscreen mode.

**Comment by bpasero** [comment]
I also notice this white flashing on Mac. For me I think it happens whenever I do loadUrl() on a browser window even if it is already visible.

**Comment by mizhigang407** [comment]
@iamdriz   Have you solve this problem? I found same problem when using nw.js,so I thought it may be chromium's trouble.

**Comment by iamdriz** [comment]
I haven't tested this in a while. But I'd assume it's still an issue if it hasn't been fixed.

**Comment by evrimoztamur** [comment]
Still not fixed indeed, having the issue even after showing the window after `did-finish-load`

**Comment by zcbenz** [comment]
Closed with #4996.
